modules/tfm/tfm/boards/board/tfm_peripherals_config.h

```diff
@@ -11,18 +11,291 @@
 extern "C" {
 #endif
 
-#ifdef SECURE_UART1
-#define TFM_PERIPHERAL_UARTE1_SECURE 1
+#include <autoconf.h>
+#include <nrfx.h>
+
+#if CONFIG_NRF_DCNF_SECURE
+#define TFM_PERIPHERAL_DCNF_SECURE                  1
+#endif
```

Comment:
enabled bool configs are defined as `1`, and disabled Kconfig settings are not in `autoconf.h`, so this should work just as well:
```suggestion
#define TFM_PERIPHERAL_DCNF_SECURE                  CONFIG_NRF_DCNF_SECURE
```

could you try that ?
I think it looks cleaner.